### PR TITLE
Dev refactor

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: client/src/main/java/connection/Client.java [hunk 1]
```diff
@@ -16,25 +18,35 @@
...
 
   private static final Logger LOGGER = Logger.getLogger(Client.class.getName());
-  private static final int PORT = 5000;
+  private static int PORT;
+  private  final String fileName ="config.properties";
 
   private Sender out;
   private Receiver in;
   private String host;
+  Connector connector;
```
Comment (sabath91) [inline client/src/main/java/connection/Client.java]: Why it isn't private?

File: client/src/main/java/gui/building/FleetDropController.java [hunk 2]
```diff
@@ -67,22 +68,22 @@
...
-   * @param client    - client instance
+   * @param client  - client instance
    * @param reactor - reactor instance
    */
 
   public FleetDropController(Client client, MessageProcessor reactor) {
     sea = new Sea();
     this.client = client;
-    this.reactor = reactor;
   }
```
Comment (sabath91) [inline client/src/main/java/gui/building/FleetDropController.java]: redundant parameters.

File: client/src/main/java/gui/starting/StartBoard.java [hunk 3]
```diff
@@ -29,9 +32,11 @@
   private static final String PLAY_BOARD_URL = "/fxmls/playBoardEmpty.fxml";
 
   private final Client client = new Client();
-  private MessageProcessor reactor;
+  private MessageProcessor processor;
   private Stage stage;
   private AnchorPane buildBoard;
+  Group playRoot;
+  FleetDropController fleetDropController;
```
Comment (sabath91) [inline client/src/main/java/gui/starting/StartBoard.java]: Some access modifiers?

File: client/src/main/java/connection/Client.java [hunk 4]
```diff
@@ -16,25 +18,35 @@
 public class Client {
 
   private static final Logger LOGGER = Logger.getLogger(Client.class.getName());
-  private static final int PORT = 5000;
+  private static int PORT;
+  private  final String fileName ="config.properties";
```
Comment (rivalez) [inline client/src/main/java/connection/Client.java]: code formatting

Comment (rivalez) [reply client/src/main/java/connection/Client.java]: imo should be static.

File: client/src/main/java/connection/Client.java [hunk 5]
```diff
@@ -16,25 +18,35 @@
...
   private String host;
+  Connector connector;
 
   /**
    * Allows the client to receive and send messages to the server.
    */
   public void run() {
     try {
-      Connector connector = SocketConnector.from(host, PORT);
+      PORT = setUpPort();
```
Comment (rivalez) [inline client/src/main/java/connection/Client.java]: why not camelCase ? that isn't const anymore

File: client/src/main/java/connection/Client.java [hunk 6]
```diff
@@ -71,4 +83,8 @@ public void sendMessage(Field field) {
   public String getMessage() {
     return in.readMessage();
   }
+
+  public boolean isGameRunning() {
```
Comment (rivalez) [inline client/src/main/java/connection/Client.java]: missing java doc

File: client/src/main/java/gui/fields/BoundField.java [hunk 7]
```diff
@@ -30,4 +32,9 @@ public int getColumn() {
   public int getRow() {
     return row;
   }
+
+  @Override
+  public void markedAsHit() {
```
Comment (rivalez) [inline client/src/main/java/gui/fields/BoundField.java]: you should rename name of this method to be verb

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 8]
```diff
@@ -20,34 +29,123 @@
...
+    winning.addEventHandler(UpdateWhenHitEvent.UPDATE, updateBoardWhenHit);
+    winning.addEventHandler(UpdateWhenMissedEvent.UPDATE_MISSED, updateBoardWhenMissed);
+    winning.addEventHandler(YourTurnEvent.TURN, enableBoard);
+    winning.addEventHandler(YouMissedEvent.WAIT, youMissed);
+    winning.addEventHandler(YouWinEvent.WIN, youWin);
+    winning.addEventHandler(YouLostEvent.LOST, youLost);
+    winning.addEventHandler(YouHitEvent.HIT, youHit);
+
+    new Thread(() -> {
+      Boolean isRunning = true;
```
Comment (rivalez) [inline client/src/main/java/gui/playing/PlayBoardController.java]: why Boolean? primitive isn't enough?

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 9]
```diff
@@ -20,34 +29,123 @@
...
+      while (isRunning) {
```
Comment (rivalez) [inline client/src/main/java/gui/playing/PlayBoardController.java]: method called initialize also checks for game finish.

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 10]
```diff
@@ -20,34 +29,123 @@
...
+
+  private void putFieldOnOpponentBoard(Field field) {
+    Platform.runLater(() -> {
+      field.markedAsHit();
+      targetBoard.getChildren().add((Node) field);
+      GridPane.setConstraints((Node) field, field.getRow(), field.getColumn());
+    });
+  }
+
+  //EVENT HANDLERS
```
Comment (rivalez) [inline client/src/main/java/gui/playing/PlayBoardController.java]: comments to remove

File: client/src/main/java/gui/playing/PlayBoardController.java [hunk 11]
```diff
@@ -20,34 +29,123 @@
...
+      targetBoard.getChildren().add(smallMast);
+      GridPane.setConstraints(smallMast, position.getColumn(), position.getRow());
+    }
+  }
+
+  private void putFieldOnOpponentBoard(Field field) {
+    Platform.runLater(() -> {
+      field.markedAsHit();
+      targetBoard.getChildren().add((Node) field);
+      GridPane.setConstraints((Node) field, field.getRow(), field.getColumn());
```
Comment (rivalez) [inline client/src/main/java/gui/playing/PlayBoardController.java]: double casting without instanceof.
you could do
Node node = (Node) filed;
and reuse it instead double casting.

File: client/src/main/java/gui/starting/StartBoard.java [hunk 12]
```diff
@@ -74,38 +78,40 @@ private void createStartBoard(Group startRoot, Scene buildScene) throws IOExcept
...
-    VBox connectPanel = (VBox) buildBoard.lookup("#connectPanel");
-    connectPanel.getChildren().add(buttonNext);
-    reactor.putObserverButtonForConnection(buttonNext);
+    Button buttonNext = (Button) buildBoard.lookup("#nextButton");
+    buttonNext.addEventHandler(ConnectEvent.CONNECT, event -> {
+      stage.setScene(playScene);
+      try {
+        createPlayBoard(playRoot, processor, fleetDropController.listOfMasts());
+      } catch (IOException e) {
+        e.printStackTrace();
```
Comment (rivalez) [inline client/src/main/java/gui/starting/StartBoard.java]: use logger

File: client/src/main/java/model/MessageProcessor.java [hunk 13]
```diff
@@ -13,42 +11,46 @@
...
-    processMessage(message);
+  private void processOpponentMissed(String message) {
+    String fieldIndex = message.split(" ")[1];
+    dispatcher.fireEvent(new UpdateWhenMissedEvent(fieldIndex));
   }
 
-  public void putObserverButtonForConnection(Button button) {
-    this.nextButton = button;
+  private void processOpponentHit(String message) {
+    String fieldIndex = message.split(" ")[1];
```
Comment (rivalez) [inline client/src/main/java/model/MessageProcessor.java]: are you sure about this? Is there validation for this ?

File: server/src/main/java/communication/ConnectionHandler.java [hunk 14]
```diff
@@ -1,44 +1,72 @@
...
 
   private void acceptPlayer(final ServerSocket serverSocket) throws IOException {
     Socket socket = serverSocket.accept();
-    playerTracker.registerPlayer(socket);
+    PlayerClient playerClient = createClient(socket);
+    playerRegistry.registerPlayer(playerClient);
+    sendConnectionConfirmationMessage(playerClient);
   }
+
+  private PlayerClient createClient(final Socket socket) throws IOException {
```
Comment (rivalez) [inline server/src/main/java/communication/ConnectionHandler.java]: how do you want to test those methods?

File: server/src/main/java/communication/ConnectionHandler.java [hunk 15]
```diff
@@ -1,44 +1,72 @@
...
+    return new PlayerClient(connectionMessage.getName(), socket, reader, playerFleet);
+  }
+
+  private void sendConnectionConfirmationMessage(PlayerClient playerClient) throws IOException {
+    final String playerIsConnected = "CON";
+    MessageSender messageSender = new MessageSender();
+    messageSender.sendMessageToPlayer(playerClient, playerIsConnected);
+  }
+
+  private ConnectionMessage prepareJSONMessage(BufferedReader reader) throws IOException {
```
Comment (rivalez) [inline server/src/main/java/communication/ConnectionHandler.java]: should be tested

File: server/src/main/java/communication/MessageSender.java [hunk 16]
```diff
@@ -1,14 +1,25 @@
...
+ *
  * @author Bartosz Pieczara/Emilia Ciastek
  * @version 1.5
  */
-class MessageSender {
-  void send(PrintWriter printWriter, String messageToSend) {
-    printWriter.println(messageToSend);
+public class MessageSender {
+
+  public void sendMessageToPlayer(PlayerClient playerClient, String messageToSend) throws IOException {
```
Comment (rivalez) [inline server/src/main/java/communication/MessageSender.java]: java doc

Comment (rivalez) [reply server/src/main/java/communication/MessageSender.java]: instead throwing IOException log it

File: server/src/main/java/engine/GameRunner.java [hunk 17]
```diff
@@ -16,46 +20,70 @@
...
+  }
+
+  private void sendMessage(final ShotResult result, Shot shot) {
     String message = result.toString();
 
+    String messageToOpponent = makeMessageForOpponent(result, shot);
+
     if (gameState == GameState.WIN) {
       message = gameState.toString();
+      messageToOpponent = "LOST";
```
Comment (rivalez) [inline server/src/main/java/engine/GameRunner.java]: don't like those magic string in your code

File: server/src/main/java/engine/GameRunner.java [hunk 18]
```diff
@@ -16,46 +20,70 @@
...
+    try {
+      messageSender.sendMessageToPlayer(playerRegistry.getCurrentPlayer(), message);
+      messageSender.sendMessageToPlayer(playerRegistry.getWaitingPlayer(), messageToOpponent);
+    } catch (IOException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage());
+    }
+  }
+
+  private String makeMessageForOpponent(ShotResult result, Shot shot) {
+    return "OPP" + result.toString() + " " + shot.asInteger();
```
Comment (rivalez) [inline server/src/main/java/engine/GameRunner.java]: use String.format(

File: utils/src/main/java/responses/Response.java [hunk 19]
```diff
@@ -0,0 +1,6 @@
+package responses;
+
+public interface Response {
```
Comment (rivalez) [inline utils/src/main/java/responses/Response.java]: java docs

**Pull request merged**